core/src/main/scala/org/apache/spark/deploy/DeployMessage.scala

```diff
@@ -200,6 +200,8 @@ private[deploy] object DeployMessages {
 
   case class KillExecutors(appId: String, executorIds: Seq[String])
 
+  case class ExecutorRemoved(appId: String, executorId: String)
```

Comment:
Why not reuse the `ExecutorUpdated` message?

Reply:
Sounds good!

Reply:
After another look, it seems we can't reuse `ExecutorUpdated` as it doesn't contain `appId`. Although we can traverse all executors to find the target one using `execId`, but it's kind of time-consuming.